datafusion-examples/examples/parquet_index.rs

```diff
@@ -0,0 +1,608 @@
+        // value column in this file
+        let row_count = stats
+            .row_count
+            .as_primitive::<UInt64Type>()
+            .iter()
+            .flatten()
+            .sum::<u64>();
+        let value_column_min = stats
+            .min
+            .as_primitive::<Int32Type>()
+            .iter()
+            .flatten()
+            .min()
+            .unwrap_or_default();
+        let value_column_max = stats
+            .max
+            .as_primitive::<Int32Type>()
+            .iter()
+            .flatten()
+            .max()
+            .unwrap_or_default();
+
+        // sanity check the statistics
+        assert_eq!(row_count, metadata.file_metadata().num_rows() as u64);
+        self.add_row(
+            file_name,
+            file_size,
+            row_count,
+            value_column_min,
+            value_column_max,
+        );
+        Ok(())
+    }
+
+    /// Add a single row values to all the in progress rows
+    fn add_row(
+        &mut self,
+        file_name: impl Into<String>,
+        file_size: u64,
+        row_count: u64,
+        value_column_min: i32,
+        value_column_max: i32,
+    ) {
+        self.filenames.push(file_name.into());
+        self.file_sizes.push(file_size);
+        self.row_counts.push(row_count);
+        self.value_column_mins.push(value_column_min);
+        self.value_column_maxs.push(value_column_max);
+    }
+
+    /// Build the index from the files added
+    fn build(self) -> Result<ParquetMetadataIndex> {
+        let Some(file_schema) = self.file_schema else {
+            return Err(internal_datafusion_err!("No files added to index"));
+        };
+
+        let index = RecordBatch::try_from_iter(vec![
+            (
+                "file_name",
+                Arc::new(StringArray::from(self.filenames)) as ArrayRef,
+            ),
+            (
+                "file_size",
+                Arc::new(UInt64Array::from(self.file_sizes)) as ArrayRef,
+            ),
+            (
+                "row_count",
+                Arc::new(UInt64Array::from(self.row_counts)) as ArrayRef,
+            ),
+            (
+                "value_column_min",
+                Arc::new(Int32Array::from(self.value_column_mins)) as ArrayRef,
+            ),
+            (
+                "value_column_max",
+                Arc::new(Int32Array::from(self.value_column_maxs)) as ArrayRef,
+            ),
+        ])?;
+
+        Ok(ParquetMetadataIndex { file_schema, index })
+    }
+}
+
+/// TODO use the new
+/// API from https://github.com/apache/datafusion/issues/10453
+pub struct ArrowStatistics {
+    /// min values
+    min: ArrayRef,
+    /// max values
+    max: ArrayRef,
+    /// Row counts (UInt64Array)
+    row_count: ArrayRef,
+    /// Null Counts (UInt64Array)
+    #[allow(dead_code)]
+    null_count: ArrayRef,
+}
+
+/// extract the minimum value in the statistics for the given column, if any
+pub fn parquet_stats_to_arrow(
```

Comment:
Here is an early prototype API of what @NGA-TRAN is working on in  https://github.com/apache/datafusion/issues/10453